tests/unity/CMakeLists.txt

```diff
@@ -0,0 +1,181 @@
+set_property(GLOBAL PROPERTY RUBY_EXECUTABLE ${RUBY_EXECUTABLE})
+
+zephyr_include_directories(
+	${CMOCK_DIR}/vendor/unity/src
+	${CMOCK_DIR}/src
+)
+
+zephyr_library_sources(
+	${CMOCK_DIR}/vendor/unity/src/unity.c
+	${CMOCK_DIR}/src/cmock.c
+)
+
+# Generate test runner file.
+function(test_runner_generate test_file_path)
+    get_property(RUBY_EXECUTABLE GLOBAL PROPERTY RUBY_EXECUTABLE)
+    get_property(CMOCK_DIR GLOBAL PROPERTY CMOCK_DIR)
+    set(UNITY_PRODUCTS_DIR ${APPLICATION_BINARY_DIR}/runner)
+
+    file(MAKE_DIRECTORY "${UNITY_PRODUCTS_DIR}")
+    get_filename_component(test_file_name "${test_file_path}" NAME)
+    set(output_file "${UNITY_PRODUCTS_DIR}/runner/runner_${test_file_name}")
+    file(MAKE_DIRECTORY "${UNITY_PRODUCTS_DIR}/runner")
+
+    execute_process(
+      COMMAND ${RUBY_EXECUTABLE}
+      ${CMOCK_DIR}/vendor/unity/auto/generate_test_runner.rb
+      ${test_file_path} ${output_file}
+      WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
+      RESULT_VARIABLE op_result
+    )
+
+    if (NOT ${op_result} EQUAL 0)
+        message(FATAL_ERROR "Failed to generate runner ${output_file}")
+    endif()
+
+    file(GLOB app_sources ${UNITY_PRODUCTS_DIR}/runner/*.c)
+    target_sources(app PRIVATE ${app_sources})
+
+    message(STATUS "Generating test runner ${unity_runner_arg3}")
+endfunction()
+
+# Generate cmock for provided header file.
+function(cmock_generate header_path dst_path)
+    get_property(RUBY_EXECUTABLE GLOBAL PROPERTY RUBY_EXECUTABLE)
+    get_property(CMOCK_DIR GLOBAL PROPERTY CMOCK_DIR)
+
+    file(MAKE_DIRECTORY "${dst_path}")
+
+    execute_process(
+      COMMAND ${RUBY_EXECUTABLE}
+      ${CMOCK_DIR}/lib/cmock.rb
+      --mock_prefix=mock_
+      --plugins=return_thru_ptr\;ignore_arg\;ignore\;callback\;array
+      --mock_path=${dst_path}
+      ${header_path}
+      WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
+      RESULT_VARIABLE op_result
+      OUTPUT_VARIABLE output_result
+    )
+
+    if (NOT ${op_result} EQUAL 0)
+          message(SEND_ERROR "${output_result}")
+          message(FATAL_ERROR "Failed to generate runner ${output_file}")
+      endif()
+endfunction()
+
+# Add --wrap linker option for each function listed in the input file.
+function(cmock_linker_trick func_name_path)
+    file(STRINGS ${func_name_path} contents)
+    set(linker_str "-Wl")
+    foreach(src ${contents})
+        set(linker_str "${linker_str},--wrap=${src}")
+    endforeach()
+    zephyr_link_libraries(${linker_str})
+endfunction()
+
+
+# Handle wrapping functions from mocked file.
+# Function takes header file and generates a file containing list of functions.
+# File is then passed to 'cmock_linker_trick' which adds linker option for each
+# function listed in the file.
+function(cmock_linker_wrap_trick header_file_path)
+    set(flist_file "${header_file_path}.flist")
+
+    execute_process(
+      COMMAND
+      ${PYTHON_EXECUTABLE}
+      ${ZEPHYR_BASE}/../nrf/scripts/unity/func_name_list.py
```

Comment:
Why not use the `${NRF_DIR}` which is being defined in the root `<nrf>/CMakeLists.txt` file ?

Reply:
Because it is not set in this scope.

We could have NRF_DIR set as a CACHE variable to get around this maybe ...

Reply:
done (https://github.com/NordicPlayground/fw-nrfconnect-nrf/pull/519/commits/44277184e3c09e27abc79329a35d83f0fd008d0b)